gwpy/io/ffldatafind.py

```diff
@@ -147,15 +149,18 @@ def _ffl_path(site, tag, basedir=None):
 
 
 @lru_cache()
-def _read_ffl(site, tag, basedir=None):
-    """Read an FFL file as a list of `CacheEntry` objects
+def _read_ffls(site, tag, basedir=None):
+    """Read all FFL files for a given site and tag
+    as a list of `CacheEntry` objects.
     """
-    ffl = _ffl_path(site, tag, basedir=basedir)
-    with open(ffl, "r") as fobj:
-        return [
-            type(entry)(site, tag, entry.segment, entry.path)
-            for entry in _iter_cache(fobj, gpstype=float)
-        ]
+    entries = []
+    for ffl in _ffl_paths(site, tag, basedir=basedir):
+        with open(ffl, "r") as fobj:
+            entries.extend(
+                type(entry)(site, tag, entry.segment, entry.path)
```

Comment:
why use type here instead of specific name?

Reply:
No good reason other than using `type` saves from importing the actual type object and using it explicitly. However, that is probably not a very good reason...

Reply:
@asouthgate, I have updated this code to use the actual type object, which is clearer.